libevm/VMFactory.cpp

```diff
@@ -175,30 +176,34 @@ po::options_description vmProgramOptions(unsigned _lineLength)
 }
 
 
-std::unique_ptr<VMFace> VMFactory::create()
+vm_ptr VMFactory::create()
 {
     return create(g_kind);
 }
 
-std::unique_ptr<VMFace> VMFactory::create(VMKind _kind)
+vm_ptr VMFactory::create(VMKind _kind)
 {
+    static const auto default_delete = [](VMFace * _vm) noexcept { delete _vm; };
+    static const auto null_delete = [](VMFace*) noexcept {};
+
     switch (_kind)
     {
 #ifdef ETH_EVMJIT
     case VMKind::JIT:
-        return std::unique_ptr<VMFace>(new EVMC{evmjit_create()});
+        return {new EVMC{evmjit_create()}, default_detete};
```

Comment:
`default_delete`

Reply:
So the CI doesn't even compile `ETH_EVMJIT` otherwise this would have been caught.

Can evmjit be compiled as a shared lib? If so, any point keeping `ETH_EVMJIT` in aleth anymore?

Reply:
Tracked in #5253.